File: PascalsTriangle2.js
const getRow = (rowIndex) => {
    let res = [];
    for(let i = 0;i <= rowIndex; i++){
        res[i] = [];
        res[i][0] = 1;
        for(let j =1 ; j < i; j++){
            res[i][j] = res[i-1][j-1] + res[i-1][j];
        }
        res[i][i] = 1;
    }

    return res[rowIndex];
};

const testGetRow = () => {
    const testCases = [
        { input: 0, expected: [1] },
        { input: 1, expected: [1, 1] },
        { input: 2, expected: [1, 2, 1] },
        { input: 3, expected: [1, 3, 3, 1] },
    ];

    testCases.forEach((testCase, index) => {
        const result = getRow(testCase.input);
        const isEqual = JSON.stringify(result) === JSON.stringify(testCase.expected);
        console.log(`Test Case ${index + 1}: ${isEqual ? 'Passed' : 'Failed'}`);
        if (!isEqual) {
            console.log(`  Expected: ${JSON.stringify(testCase.expected)}`);
            console.log(`  Received: ${JSON.stringify(result)}`);
        }
    });
};

testGetRow();

File: pascals_triangle_2.py
def get_row(row_index):
    res = []
    for i in range(row_index + 1):
        res.append(1)
        for j in range(i - 1, 0, -1):
            res[j] = res[j - 1] + res[j]
    return res

def test_get_row():
    test_cases = [
        {"input": 0, "expected": [1]},
        {"input": 1, "expected": [1, 1]},
        {"input": 2, "expected": [1, 2, 1]},
        {"input": 3, "expected": [1, 3, 3, 1]}
    ]

    for index, test_case in enumerate(test_cases):
        result = get_row(test_case["input"])
        if result == test_case["expected"]:
            print(f"Test Case {index + 1}: Passed")
        else:
            print(f"Test Case {index + 1}: Failed")
            print(f"  Expected: {test_case['expected']}")
            print(f"  Received: {result}")

test_get_row()